dsl/camel-jbang/CamelJBang.java

```diff
@@ -0,0 +1,249 @@
+        private File lockFile;
+
+        public ShutdownRoute(File lockFile) {
+            this.lockFile = lockFile;
+        }
+
+        public void configure() {
+            fromF("file-watch://%s?events=DELETE&antInclude=%s", lockFile.getParent(), lockFile.getName())
+                .process(p -> context.shutdown());
+        }
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        switch (debugLevel) {
+            case "trace": Configurator.setRootLevel(Level.TRACE); break;
+            case "debug": Configurator.setRootLevel(Level.DEBUG); break;
+            case "info": Configurator.setRootLevel(Level.INFO); break;
+            case "warn": Configurator.setRootLevel(Level.WARN); break;
+            case "fatal": Configurator.setRootLevel(Level.FATAL); break;
+            default: {
+                System.err.println("Invalid debug level " + debugLevel);
+                return 1;
+            }
+        }
+
+        File bindingFile = new File(binding);
+        if (!bindingFile.exists()) {
+            System.err.println("The binding file does not exist");
+
+            return 1;
+        }
+
+        System.setProperty("camel.main.routes-include-pattern", "file:" + binding);
+        System.setProperty("camel.main.name", "CamelJBang");
+
+        System.out.println("Starting Camel JBang!");
+        KameletMain main = new KameletMain();
+
+        main.configure().addRoutesBuilder(new ShutdownRoute(createLockFile()));
+        main.start();
+        context = main.getCamelContext();
+
+        main.run();
+        return 0;
+    }
+
+    public File createLockFile() throws IOException {
+        File lockFile = File.createTempFile(".run", ".camel.lock", new File("."));
+
+        System.out.printf("A new lock file was created on %s. Delete this file to stop running%n",
+            lockFile.getAbsolutePath());
+        lockFile.deleteOnExit();
+
+        return lockFile;
+    }
+}
+
+@Command(name = "search", description = "Search for a Kamelet in the Kamelet catalog")
+class Search implements Callable<Integer> {
+
+    @Option(names = {"--search-term"}, defaultValue = "", description = "Default debug level")
+    private String searchTerm;
+
+    private void downloadResource(File indexFile) throws Exception {
+        KameletMain main = new KameletMain();
+        main.start();
+        CamelContext context = main.getCamelContext();
+
+        try (GitHubResourceResolver resolver = new GitHubResourceResolver()) {
+            resolver.setCamelContext(context);
+
+            Resource resource = resolver.resolve("github:apache:camel-kamelets:docs/modules/ROOT/nav.adoc");
+
+            if (!resource.exists()) {
+                throw new Exception("The resource does not exist");
+            }
+
+            try (FileWriter w = new FileWriter(indexFile)) {
+                Reader resourceReader = resource.getReader();
+
+                resourceReader.transferTo(w);
```

Comment:
This API is Java 10+, so we cannot use it for the moment. I'll have to use a different one.